### DISCUSSION TITLE
Automation of the gitlab variable configuration

### DISCUSSION THREAD

**Discussion by Guits32** [firecow/gitlab-ci-local#912] [Ideas]
To simplify my job with your awesome tool (thank a lot guys!) I develop a little **bash** script. The goal is to retrieve gitlab CI/CD variables and store them in the gitlab-ci-local config. Here is the usage:

``` shell
$ - gcl-config 
Usage:       gcl-config
Version:     1.0
Description: Add to "gitlab-ci-local" configuration the remote gitlab CI/CD variables of a specific group/project (you can find the generated configuration file here - ~/.gitlab-ci-local/variables.yml)
Examples:
  gcl-config -p project_id      : retrieve CI variables assciated to this project and update gitlab-ci-local config
  gcl-config -g group_id        : retrieve CI variables assciated to this group and update gitlab-ci-local config

Options:
  -h|--help                               : Display this help message
  -v|--verbose                         : Display DEBUG log messages
  -p|--project                          : Define the id of the project to search on
  -g|--group                            : Define the id of the group to search on
  -s|--show-config                 : print the current gitlab-ci-local config
```

The idea is quite simple: when I work on a CI I need the variables of the project itself and the variables of the parent groups. To do this, I run `gcl-config` specifying whether it is a `-p` project or a `-g` group and give it the id. Next, the script fetches the gitlab API to retrieve the CI/CD variables, and finally refactors the JSON API response to match the gitlab-ci-local yaml configuration.

I know it's bash, and you are developing this tool in TS, but if you are interrested in any way let me know, and I will share my code.

**Comment by firecow** [comment]
You need maintainer on the group and project you want variables from right?

**Reply by Guits32** [reply]
Yes right

**Reply by firecow** [reply]
Hmmm... 

I'm not sure I like the idea of devs fetching potential stage/prod secrets from a groups or projects variables.

But lets keep this script here in case someone wants to develop like this 😁

**Reply by Guits32** [reply]
Here it is: [gcl-config.zip](https://github.com/firecow/gitlab-ci-local/files/11891431/gcl-config.zip)

There is one specific config which is not dynamic: the gitlab domain.

The variable is defined in `_config.sh` line 3, you can change it to feet your needs.
